pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -235,9 +236,14 @@ func (r *{{ .Resource.Kind }}Reconciler) doFinalizerOperationsFor{{ .Resource.Ki
 }
 
 // deploymentFor{{ .Resource.Kind }} returns a {{ .Resource.Kind }} Deployment object
-func (r *{{ .Resource.Kind }}Reconciler) deploymentFor{{ .Resource.Kind }}({{ lower .Resource.Kind }} *{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}) *appsv1.Deployment {
+func (r *{{ .Resource.Kind }}Reconciler) deploymentFor{{ .Resource.Kind }}(ctx context.Context, {{ lower .Resource.Kind }} *{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}) *appsv1.Deployment {
 	ls := labelsFor{{ .Resource.Kind }}({{ lower .Resource.Kind }}.Name)
 	replicas := {{ lower .Resource.Kind }}.Spec.Size
+	log := log.FromContext(ctx)
+	image, err := imageFor{{ .Resource.Kind }}()
+	if err != nil {
+    	log.Error(err, "unable to get image for {{ .Resource.Kind }}")
+	}
```

Comment:
Why do we need to get the image here?
Why it cannot be done inside of the method that creates the deployment?

Also, note that the first operation must to be check if the cr exists otherwise has no reason to do anything else

Reply:
Just confirming. So, we should add another argument in the `deploymentFor` function for the `image` and execute this code in the `Reconcile` function before we execute the `deploymentFor` function.

Reply:
Oh it is in the deployment method. That was my mistake, for some reason I thought that it was in the reconcile.